api/src/main/java/org/eclipse/microprofile/metrics/annotation/NamedMetadata.java

```diff
@@ -0,0 +1,54 @@
+/*
+ **********************************************************************
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICES file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ **********************************************************************/
+package org.eclipse.microprofile.metrics.annotation;
+
+import org.eclipse.microprofile.metrics.MetricType;
+import org.eclipse.microprofile.metrics.MetricUnits;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+@Repeatable(NamedMetadatas.class)
+public @interface NamedMetadata {
+
+    // TODO: javadoc comments
+
+    String name();
+
+    String metricName();
```

Comment:
are we assuming that names defined this say have to be absolute?

Reply:
Well, we want this to map to the same name for every metric on which the annotation is applied, so we can't prepend the package/class to each metric, that would mean each using metric has a different name. So yes this would mean the name is absolute.
Unless we actually really want to prepend the package/class to each metric separately, but then we would lose one big advantage, and that is having a 1:1 mapping between a `@NamedMetadata` annotation instance and an actual `Metadata` object at runtime

Reply:
I agree - I think the metric names should always be considered absolute when specified using `@NamedMetadata`.  Hard to come up with any useful reason for the name to be relative.